### DISCUSSION TITLE
Is there a set of constants for unit conversion?

### DISCUSSION THREAD

**Discussion by jrom99** [pymupdf/PyMuPDF#1654] [Q&A] [answered]
I want to draw a circle with a given radius, is there a function or constant available for use? Currently I"m using this:

```python
def as_dist(v, unit="mm"):
    p0 = fitz.Point(0,0)
    p1 = fitz.Point(1,0)
    unitary_dist = 1/(p0.distance_to(p1, unit))
    return v*unitary_dist

mm = as_dist(1, "mm")
```

Is there some constant or function within pymupdf itself for this?
```python
from fitz import mm
Page.draw_circle(center, radius=2*mm)
# or
from fitz import as_dist
Page.draw_circle(center, radius=as_dist(2, "mm"))
```

**Comment by JorjMcKie** [comment, marked as answer]
No, there is not.
Please simply use 1 inch = 72 points = 25.4 mm

**Comment by JorjMcKie** [comment]
PyMuPDF views these types of constants as parts of some general module.
Would be a never ending story, too. Questions for area units would not be farfteched, too, would they.
But there are standard page sizes recorded like Letter or ISO-A4: `fitz.paper_size()` and friends.

**Reply by jrom99** [reply]
I see, thanks for the info. I fail to see use cases that would need area constants, though. (I can't remember any function in pymupdf that accepts an area as an argument, are there many of them?)

As for units, I think mm, cm and inch would cover enough use cases to justify being the only ones defined, especially when dealing with paddings and margins.

**Reply by JorjMcKie** [reply]
> accepts an area as an argument

Well, I have been alluding to the various page rectangles primarily. Rectangles / bboxes  appear all over the place in PyMuPDF. They **_always_** are defined in `point` units (1 in = 72 pt).
At least in PDF, "real world" units like inch, foot, yard, m, cm, mm play no role at all, and I believe you will not find a single referral to them in the PDF spec ... and neither in MuPDF.

Some examples for standard page / rectangle sizes (I recall there are more than a dozen):
```python
import fitz
fitz.paper_size("a4")
(595, 842)
fitz.paper_size("letter")
(612, 792)
fitz.paper_rect("a4")
Rect(0.0, 0.0, 595.0, 842.0)
fitz.paper_rect("letter")
Rect(0.0, 0.0, 612.0, 792.0)
```
Whether you insert an image or an annotation, and many other things: rectangles are the  most frequent arguments. Similarly for extracting stuff.

There are a select few functions in PyMuPDF which return other units than points:
```python
r = fitz.paper_rect("a4")
r.get_area("cm")  # other options: "mm" = millimeter, "in" = inch, "px" = point (default)
623.4928788580246
p1 = fitz.Point(0,0)
p2 = fitz.Point(1,1)
p1.distance_to(p2, "cm")
0.04989031178371752
p1.distance_to(p2, "in")
0.019641855032959652
p1.distance_to(p2, "px")
1.4142135623730951
p1.distance_to(p2, "mm")
0.49890311783717517
```
I think the above should serve your purpose.

**Comment by JorjMcKie** [comment]
But if you do have a PR that implements a function similar to `fitz.from_unit(2, "cm")`, I would welcome it and integrate it in the next version.
The example would mean that 2 is given in centimeters and should be converted to "pt".
